### PR TITLE
Add About me patterns

### PR DESCRIPTION
**Pull request by scruffian**
**Description**
This adds the patterns from https://github.com/WordPress/twentytwentytwo/issues/15

**Screenshots**
localhost_4759__p=559

**Testing Instructions** 
1. Add each of the patterns from the block inserter
2. You will need to set a site logo and have some pages to see all aspects of the patterns

### CODE REVIEW

**Comment by MaggieCabrera** [comment]
I think we should add .DS_Store to gitignore :D

File: inc/patterns/about-me-big-image-and-buttons.php
Comment (MaggieCabrera) [inline inc/patterns/about-me-big-image-and-buttons.php]: this needs an alt text

File: inc/patterns/about-me-link-in-bio-dark.php
Comment (MaggieCabrera) [inline inc/patterns/about-me-link-in-bio-dark.php]: missing alt text here too

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (MaggieCabrera) [inline inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: this is not escaped, we can do it like so:
```suggestion
					<h2 style="font-size:72px;font-weight:300;line-height:1.115"><em>' . wp_kses_post(  __( 'Jesús<br>Rodriguez', 'twentytwentytwo' ) ) . '</em></h2>
```

File: inc/patterns/about-me-link-in-bio-dark.php
Comment (MaggieCabrera) [reply inc/patterns/about-me-link-in-bio-dark.php]: there's a couple more, you get my drift :D

File: inc/patterns/about-me-big-image-and-buttons.php
Comment (jffng) [inline inc/patterns/about-me-big-image-and-buttons.php]: When you set a width on a button, they don't wrap nicely on smaller screens: 

Screen Shot 2021-10-14 at 10 13 58 AM

It's because of this line:

Screen Shot 2021-10-14 at 10 15 20 AM


It seems like this needs better defaults in Gutenberg, whether that's through `flex-basis`, `min-width` or otherwise.

File: inc/patterns/about-me-big-image-and-buttons.php
Comment (jffng) [inline inc/patterns/about-me-big-image-and-buttons.php]: Is it possible to use the preset here? 
```suggestion
					<!-- wp:social-links {"iconColor":"primary","iconColorValue":"var(--wp--preset--color--primary)","className":"is-style-logos-only","layout":{"type":"flex","justifyContent":"center"}} -->
```

File: inc/block-patterns.php
Comment (jffng) [inline inc/block-patterns.php]: I wonder if these need their own section, or could be included in the Page Layouts being added in #104. If their own section, nitpick on the capitalization: 
```suggestion
				array( 'label' => __( 'Twenty Twenty-Two: About Me', 'twentytwentytwo' ) )
```

File: inc/patterns/about-me-left-media-and-text-with-logo-header-paragraph-and-social.php
Comment (jffng) [inline inc/patterns/about-me-left-media-and-text-with-logo-header-paragraph-and-social.php]: The top and bottom padding for the content looks a little tight compared to the comps:

Screen Shot 2021-10-14 at 10 27 57 AM

On smaller screens, the padding collapses entirely: 

Top | Bottom
------- | ------
Screen Shot 2021-10-14 at 10 28 13 AM | Screen Shot 2021-10-14 at 10 28 21 AM

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (jffng) [inline inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: Should the nav have the responsive overlay option enabled on mobile?

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (jffng) [inline inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: Could this use responsive padding values so it scales nicely on mobile? 

Screen Shot 2021-10-14 at 10 34 46 AM

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (jffng) [inline inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: I think these should be a responsive value per @kjellr's comment here: https://github.com/WordPress/twentytwentytwo/pull/104#discussion_r728904778

File: inc/patterns/about-me-right-media-and-text-with-logo-header-paragraph-and-social.php
Comment (jffng) [inline inc/patterns/about-me-right-media-and-text-with-logo-header-paragraph-and-social.php]: Same padding issues as "Left Media & Text with Logo, Header, Paragraph, and Social".

File: inc/patterns/about-me-right-media-and-text-with-logo-header-paragraph-and-social.php
Comment (jffng) [inline inc/patterns/about-me-right-media-and-text-with-logo-header-paragraph-and-social.php]: Same comment about responsive value here.

**Review by jffng — COMMENTED** [review]
Thanks for working on this!

File: inc/block-patterns.php
Comment (kjellr) [reply inc/block-patterns.php]: I think this is fine for now (with the capital M). We can move them over to Page Layouts after if we'd like.

File: inc/patterns/about-me-big-image-and-buttons.php
Comment (kjellr) [reply inc/patterns/about-me-big-image-and-buttons.php]: Yeah, this needs to be manually swapped out until https://github.com/WordPress/gutenberg/issues/35480 is resolved.

**Comment by kjellr** [comment]
I just pushed responsive values for the text. I have some ideas for how to improve responsive behavior in general for some of these — I'll push some additional commits for that in a little bit.

File: inc/patterns/about-me-big-image-and-buttons.php
Comment (kjellr) [reply inc/patterns/about-me-big-image-and-buttons.php]: I recreated this one using columns, and it works much better.... except at middle breakpoints. I'm not sure what else to try. 😕 Might end up needing to revise this design. 
buttons

**Comment by kjellr** [comment]
Ok I pushed a bunch of responsive cleanups. I also added alt text and did some general cleanup wherever I saw things looking weird. Should be ok for another review.

File: inc/block-patterns.php
Comment (jffng) [inline inc/block-patterns.php]: The other pattern categories don't have colons:

Screen Shot 2021-10-14 at 3 06 08 PM

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (jffng) [inline inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: Getting some strange behavior with this one.

1. It appears black (primary) in the editor and grey on the front-end:

Editor | Frontend
------- | --------
Screen Shot 2021-10-14 at 3 13 27 PM | Screen Shot 2021-10-14 at 3 13 40 PM

2. The cover block encounters an error when I try to select it in the editor: 

Screen Shot 2021-10-14 at 3 10 44 PM

The console error thrown is an unclear react one: 

```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.
```

I'm not sure either of these issues are related to the PR, or a Gutenberg issue. 2 happens when I insert any cover block. Running trunk.

**Comment by kjellr** [comment]
I've also updated `readme.txt` with the credit lines for these images.

**Review by jffng — COMMENTED** [review]
Really nice work with these, they are beautiful. 

A couple comments, but I think these are ready to come in pending that issue with the cover block.

File: inc/block-patterns.php
Comment (kjellr) [reply inc/block-patterns.php]: I think we should probably actually just place these in the "Pages" category that will be defined when #104 lands, so let's wait for that and just change this before we merge.

File: inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php
Comment (kjellr) [reply inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: That's really weird — I can't replicate either of those issues. I'm also running Gutenberg `trunk`.

Comment (jffng) [reply inc/patterns/about-me-logo-header-description-and-social-on-left-menu-and-separator-on-right.php]: I just rebuilt trunk and it went away 🤷

**Review by jffng — APPROVED** [review]

**Pull request merged**
